adaptive_breakpoints/lib/src/adaptive_breakpoints.dart

```diff
@@ -0,0 +1,377 @@
+    window: AdaptiveWindow.xs,
+    columns: 4,
+    margins: 16.0,
+    gutters: 16.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(360, 399),
+    portrait: 'medium handset',
+    window: AdaptiveWindow.xs,
+    columns: 4,
+    margins: 16.0,
+    gutters: 16.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(400, 479),
+    portrait: 'large handset',
+    window: AdaptiveWindow.xs,
+    columns: 4,
+    margins: 16.0,
+    gutters: 16.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(480, 599),
+    portrait: 'large handset',
+    landscape: 'small handset',
+    window: AdaptiveWindow.xs,
+    columns: 4,
+    margins: 16.0,
+    gutters: 16.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(600, 719),
+    portrait: 'small tablet',
+    landscape: 'medium handset',
+    window: AdaptiveWindow.s,
+    columns: 8,
+    margins: 16.0,
+    gutters: 16.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(720, 839),
+    portrait: 'large tablet',
+    landscape: 'large handset',
+    window: AdaptiveWindow.s,
+    columns: 8,
+    margins: 24.0,
+    gutters: 24.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(840, 959),
+    portrait: 'large tablet',
+    landscape: 'large handset',
+    window: AdaptiveWindow.s,
+    columns: 12,
+    margins: 24.0,
+    gutters: 24.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(960, 1023),
+    landscape: 'small tablet',
+    window: AdaptiveWindow.s,
+    columns: 12,
+    margins: 24.0,
+    gutters: 24.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(1024, 1279),
+    landscape: 'large tablet',
+    window: AdaptiveWindow.m,
+    columns: 12,
+    margins: 24.0,
+    gutters: 24.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(1280, 1439),
+    landscape: 'large tablet',
+    window: AdaptiveWindow.m,
+    columns: 12,
+    margins: 24.0,
+    gutters: 24.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(1440, 1599),
+    portrait: 'small handset',
+    window: AdaptiveWindow.l,
+    columns: 12,
+    margins: 24.0,
+    gutters: 24.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(1600, 1919),
+    portrait: 'small handset',
+    window: AdaptiveWindow.l,
+    columns: 12,
+    margins: 24.0,
+    gutters: 24.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(1920, 10000000),
+    portrait: 'small handset',
+    window: AdaptiveWindow.xl,
+    columns: 12,
+    margins: 24.0,
+    gutters: 24.0,
+  ),
+];
+
+/// Returns ths Window Size to the user. This is useful when the user wants to
+/// compare the MediaQuery to the current window size.
+AdaptiveWindow getWindowSize(Size size) {
+  double width = size.width;
+  for (BreakpointSystemEntry entry in breakpointSystem) {
+    if (entry.range.start <= width && width <= entry.range.end) {
+      return entry.window;
+    }
+  }
+  throw AssertionError('WindowSize is too big');
+}
+
+BreakpointSystemEntry getBreakpointEntry(Size size) {
```

Comment:
The primary API should maybe take in a context (simplify the call on the package user end), with an option to (either a separate function or a diff param) to call it with a Size

Reply:
Done